runner/debug/consul.go

```diff
@@ -0,0 +1,107 @@
+package debug
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/hashicorp/hcdiag/op"
+	"github.com/hashicorp/hcdiag/redact"
+	"github.com/hashicorp/hcdiag/runner"
+)
+
+var _ runner.Runner = ConsulDebug{}
+
+// ConsulDebugConfig is a config struct for ConsulDebug runners
+type ConsulDebugConfig struct {
+	Archive    string
+	Duration   string
+	Interval   string
+	Captures   []string
+	Redactions []*redact.Redact
+}
+
+// ConsulDebug represents a ConsulDebug runner
+type ConsulDebug struct {
+	Archive    string           `json:"archive"`
+	Duration   string           `json:"duration"`
+	Interval   string           `json:"interval"`
+	Captures   []string         `json:"captures"`
+	Redactions []*redact.Redact `json:"redactions"`
+
+	output string
+}
+
+func (d ConsulDebug) ID() string {
+	return "ConsulDebug"
+}
+
+func NewConsulDebug(cfg ConsulDebugConfig, tmpDir string, debugDuration time.Duration, debugInterval time.Duration) (*ConsulDebug, error) {
+	dbg := ConsulDebug{
+		// No compression because the hcdiag bundle will get compressed anyway
+		Archive: "true",
+		// Use debug duration and interval
+		Duration: debugDuration.String(),
+		Interval: debugInterval.String(),
+		// Creates a subdirectory inside output dir
+		output:     tmpDir,
+		Captures:   cfg.Captures,
+		Redactions: cfg.Redactions,
+	}
+
+	if cfg.Archive != "" {
+		dbg.Archive = cfg.Archive
+	}
+
+	if len(cfg.Duration) > 0 {
```

Comment:
Since the other similar checks we've done have compared to the empty string, I think we should do that here too.

Reply:
I updated this while Dave is away.